src/main/java/com/pretz/dyntreeserver/service/UserService.java

```diff
@@ -0,0 +1,69 @@
+package com.pretz.dyntreeserver.service;
+
+import com.pretz.dyntreeserver.domain.User;
+import com.pretz.dyntreeserver.repository.UserRepo;
+import com.pretz.dyntreeserver.service.exceptions.IncorrectPasswordException;
+import com.pretz.dyntreeserver.service.exceptions.UserCreationFailException;
+import com.pretz.dyntreeserver.service.exceptions.UserNotFoundException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+
+import static com.pretz.dyntreeserver.service.Seed.SEED;
+
+@Service
+public class UserService {
+
+    private static final long TOKEN_VALID_TIME = 3600 * 1000; //1 hour in ms
+
+    private UserRepo userRepo;
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(UserRepo userRepo, PasswordEncoder passwordEncoder) {
+        this.userRepo = userRepo;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public String validateUser(User user) {
+        if (userRepo.findByName(user.getName()).isPresent()) {
+            if (passwordEncoder.matches(user.getPassword(), userRepo.getOne(user.getId()).getPassword())) {
+                return createToken(user);
+            } else {
+                throw new IncorrectPasswordException(user);
```

Comment:
One more suggestion here. As described in the [OWASP Auth Cheatsheet](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Authentication_and_Error_Messages), it's a bad practice to inform the user in such a explicit way (e.g. "incorrect password for user XY").

It makes it easier for possible attackers to do password bruteforce attacks as they already know if the login is correct.
As described in the article I've provided, it's preferred to return sth like `incorrect username or password`

Reply:
Ok, thanks for the tip.